app/models/concerns/hostext/search.rb

```diff
@@ -198,10 +196,11 @@ def search_by_proxy(key, operator, value)
       end
 
       def search_cast_facts(key, operator, value)
-        table_id = self.fact_values_table_counter = (self.fact_values_table_counter || 0) + 1
+        in_query = Host.joins(:fact_values => :fact_name).select(:id).
+                    where("#{FactName.table_name}.name = ?", key.split('.')[1]).
+                    where(cast_facts(FactValue.table_name, key, operator, value)).to_sql
```

Comment:
can you verify that when searching for multiple facts, the correct joins are performed? e.g.

FactName1 = A and FactValue1=X OR FactName2 = B and FactValue2=Y ?

I have a distinct memory that when doing multiple joins on fact values, the conditions has to be correct, otherwise you would get things like
FactName = A and Value = Y mixed up etc.

Reply:
Added some tests to verify this case.